packages/addons-inject/tests/wrangler.test.js

```diff
@@ -0,0 +1,256 @@
+import { fetchMock, SELF } from "cloudflare:test";
+import { afterEach, beforeAll, describe, expect, it } from "vitest";
+
+import { AddonsConstants } from "../index";
+
+beforeAll(() => {
+  // Enable outbound request mocking...
+  fetchMock.activate();
+  // ...and throw errors if an outbound request isn't mocked
+  fetchMock.disableNetConnect();
+});
+
+// Ensure we matched every mock we defined
+afterEach(() => fetchMock.assertNoPendingInterceptors());
+
+describe("Addons when disabled", async () => {
+  it("skips injection on HTML based on headers", async () => {
+    fetchMock
+      .get("http://test-builds.devthedocs.org")
+      .intercept({ path: "/en/latest/" })
+      .reply(200, "<html></html>", {
+        headers: {
+          "Content-type": "text/html",
+          "X-RTD-Force-Addons": false,
+          "X-RTD-Hosting-Integrations": false,
+        },
+      });
+    let response = await SELF.fetch(
```

Comment:
Is this `SELF.fetch()` the "magic" that executes our worker? I'm not sure to understand _where_ the CF worker is executed in these tests.

Reply:
> Also, where we indicate that our CF worker lives in the index.js?
 
This is the `wrangler.toml` that dictates this.

> Is this SELF.fetch() the "magic" that executes our worker?

Yup, this executes `wrangler dev` pretty much. You can read more on this pattern here:

https://developers.cloudflare.com/workers/testing/integration-testing/